pandas/tests/indexes/timedeltas/test_arithmetic.py

```diff
@@ -28,23 +29,98 @@ def freq(request):
 class TestTimedeltaIndexArithmetic(object):
     _holder = TimedeltaIndex
 
-    @pytest.mark.xfail(reason='GH#18824 ufunc add cannot use operands...')
-    def test_tdi_with_offset_array(self):
+    @pytest.mark.parametrize('box', [np.array, pd.Index])
+    def test_tdi_add_offset_array(self, box):
         # GH#18849
-        tdi = pd.TimedeltaIndex(['1 days 00:00:00', '3 days 04:00:00'])
-        offs = np.array([pd.offsets.Hour(n=1), pd.offsets.Minute(n=-2)])
-        expected = pd.TimedeltaIndex(['1 days 01:00:00', '3 days 04:02:00'])
+        tdi = TimedeltaIndex(['1 days 00:00:00', '3 days 04:00:00'])
+        other = np.array([pd.offsets.Hour(n=1), pd.offsets.Minute(n=-2)])
 
-        res = tdi + offs
+        expected = TimedeltaIndex([tdi[n] + other[n] for n in range(len(tdi))],
+                                  name=tdi.name, freq='infer')
+
+        with tm.assert_produces_warning(PerformanceWarning):
+            res = tdi + other
         tm.assert_index_equal(res, expected)
 
-        res2 = offs + tdi
+        with tm.assert_produces_warning(PerformanceWarning):
+            res2 = other + tdi
         tm.assert_index_equal(res2, expected)
 
-        anchored = np.array([pd.offsets.QuarterEnd(),
-                             pd.offsets.Week(weekday=2)])
+        anchored = box([pd.offsets.QuarterEnd(),
+                        pd.offsets.Week(weekday=2)])
+        with pytest.raises(TypeError):
+            with tm.assert_produces_warning(PerformanceWarning):
+                tdi + anchored
+        with pytest.raises(TypeError):
+            with tm.assert_produces_warning(PerformanceWarning):
+                anchored + tdi
+
```

Comment:
ideally could parameterize some of these for index, ndarray & Series as well (you can leave them here is ok). I believe you have some duplication in test_base

Reply:
Yah, part of it is tricky because parametrizing with `box` and `names` at the same time is cumbersome.  My thought at the moment is to get the bugs fixed and tested first and take on refactoring later (when there's not a constant flow of rebasing to be done).